qiita_db/metadata_template.py

```diff
@@ -1672,16 +1686,29 @@ def add_category(self, category, samples_and_values, dtype, default):
             If the column already exists
         """
         table_name = self._table_name(self.study_id)
-        conn_handler = SQLConnectionHandler()
 
         if category in self.categories():
             raise QiitaDBDuplicateError(category, "N/A")
-
-        conn_handler.execute("""
-            ALTER TABLE qiita.{0}
-            ADD COLUMN {1} {2}
-            NOT NULL DEFAULT '{3}'""".format(table_name, category, dtype,
-                                             default))
+        conn_handler = SQLConnectionHandler()
+        queue = 'add_col_%d' % self._id
+        conn_handler.create_queue(queue)
+
+        # add the column to the sample template
+        sql = """ALTER TABLE qiita.{0}
+            ADD COLUMN {1} {2}""".format(table_name, category, dtype)
+        if default is not None:
+            # mogrify the sql and add to end
+            with conn_handler.get_postgres_cursor() as cur:
+                sql = sql + cur.mogrify(" NOT NULL DEFAULT %s", [default])
```

Comment:
Check the comment of the first answer on this [thread](http://stackoverflow.com/q/6775497/3746629). In short, mogrify should be used only for debugging, not for generating code. Anyways, I also find this solution more convoluted, so if you can just add the if and generate either of the sql in the specific case of the if would be more cleaner/maintainable.